db.go

```diff
@@ -50,13 +50,13 @@ func (u AuthUserType) CanRenameIdentity() bool {
-	SetPassword(userId UserId, password string) error                                                                              // This sets the password to a user account
-	ResetPasswordStart(userId UserId, expires time.Time) (string, error)                                                           // Create a one-time token that can be used to reset the password with a subsequent call to ResetPasswordFinish
-	ResetPasswordFinish(userId UserId, token string, password string) error                                                        // Check that token matches the last one generated by ResetPasswordStart, and if so, call SetPassword
-	CreateIdentity(email, username, firstname, lastname, mobilenumber, password string, authUserType AuthUserType) (UserId, error) // Create a new identity. If the identity already exists, then this must return ErrIdentityExists.
-	UpdateIdentity(userId UserId, email, username, firstname, lastname, mobilenumber string, authUserType AuthUserType) error      // Update an identity. Change email address or name etc.
-	ArchiveIdentity(userId UserId) error                                                                                           // Archive an identity
+	Authenticate(identity, password string) error                                                                                                                                                                                // Return nil error if the username and password are correct, otherwise one of ErrIdentityAuthNotFound or ErrInvalidPassword
+	SetPassword(userId UserId, password string) error                                                                                                                                                                            // This sets the password to a user account
+	ResetPasswordStart(userId UserId, expires time.Time) (string, error)                                                                                                                                                         // Create a one-time token that can be used to reset the password with a subsequent call to ResetPasswordFinish
+	ResetPasswordFinish(userId UserId, token string, password string) error                                                                                                                                                      // Check that token matches the last one generated by ResetPasswordStart, and if so, call SetPassword
+	CreateIdentity(email, username, firstname, lastname, mobilenumber, telephonenumber, remarks string, created int64, createdby string, modified int64, modifiedby, password string, authUserType AuthUserType) (UserId, error) // Create a new identity. If the identity already exists, then this must return ErrIdentityExists.
```

Comment:
I think we should separate the non-essential fields out into a "OptionalUserFields" struct. These would include:
* firstname
* lastname
* mobilenumber
* telephonenumber
* remarks
* created
* createdby
* modified
* modifiedby

Or, perhaps, just take an `AuthUser` struct, and do reasonable things like ignore UserId when creating a new identity.